tools/api-markdown-documenter/src/api-item-transforms/TransformApiModel.ts

```diff
@@ -0,0 +1,270 @@
+			);
+		}
+
+		if (packageEntryPoints.length === 1) {
+			// If a package only contains a single entry-point, we will bubble up the entry-point's contents
+			// directly into the package-level document.
+
+			const entryPoint = packageEntryPoints[0];
+
+			documents.set(
+				packageItem,
+				createDocumentForSingleEntryPointPackage(packageItem, entryPoint, config),
+			);
+
+			const packageDocumentItems = getDocumentItems(entryPoint, config);
+			for (const apiItem of packageDocumentItems) {
+				if (!documents.has(apiItem)) {
+					documents.set(apiItem, apiItemToDocument(apiItem, config));
+				}
+			}
+		} else {
+			// If a package contains multiple entry-points, we will generate a separate document for each.
+			// The package-level document will enumerate the entry-points.
+
+			documents.set(
+				packageItem,
+				createDocumentForMultiEntryPointPackage(packageItem, config),
+			);
+
+			for (const entryPoint of packageEntryPoints) {
+				documents.set(entryPoint, createDocumentForApiEntryPoint(entryPoint, config));
+
+				const packageDocumentItems = getDocumentItems(entryPoint, config);
+				for (const apiItem of packageDocumentItems) {
+					if (!documents.has(apiItem)) {
+						documents.set(apiItem, apiItemToDocument(apiItem, config));
+					}
+				}
+			}
+		}
+	}
+
+	logger.success("Documents generated!");
+
+	return [...documents.values()];
+}
+
+/**
+ * Walks the provided API item's member tree and reports all API items that should be rendered to their own documents.
+ *
+ * @param apiItem - The API item in question.
+ * @param config - See {@link ApiItemTransformationConfiguration}
+ */
+function getDocumentItems(
+	apiItem: ApiItem,
+	config: Required<ApiItemTransformationConfiguration>,
+): ApiItem[] {
+	const { documentBoundaries } = config;
+
+	const result: ApiItem[] = [];
+	for (const childItem of apiItem.members) {
+		if (doesItemRequireOwnDocument(childItem, documentBoundaries)) {
+			result.push(childItem);
+		}
+		result.push(...getDocumentItems(childItem, config));
+	}
+	return result;
+}
+
+/**
+ * Generates a {@link DocumentNode} for the specified `apiModel`.
+ *
+ * @param apiModel - The API model content to be rendered. Represents the root of the API suite.
+ * @param config - See {@link MarkdownDocumenterConfiguration}.
+ *
+ * @returns The rendered Markdown document.
+ */
+export function createDocumentForApiModel(
+	apiModel: ApiModel,
+	config: Required<ApiItemTransformationConfiguration>,
+): DocumentNode {
+	const { logger, transformApiModel: createModelBodySections } = config;
+
+	logger.verbose(`Generating API Model document...`);
+
+	// Note: We don't render the breadcrumb for Model document, as it is always the root of the file hierarchical
+
+	// Render body contents
+	const sections = createModelBodySections(apiModel, config);
+
+	logger.verbose(`API Model document rendered successfully.`);
+
+	return createDocument(apiModel, sections, config);
+}
+
+/**
+ * TODO
```

Comment:
For this PR?

Reply:
there are some other TODOs elsewhere, same question.

Reply:
Ah, yup. Need to fill those in.